lib/widgets/main_screen_bottomnavbar.dart

```diff
@@ -0,0 +1,132 @@
+        ),
+      ),
+      height: 70,
+      padding: const EdgeInsets.only(
+        top: 10,
+        left: 10,
+        right: 10,
+        bottom: 0,
+      ),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+        children: [
+          _SelectedIcon(
+            image: 'assets/custom_svg_icons/home.svg',
+            index: 0,
+            label: 'Home',
+            onTap: onTap,
+          ),
+          _SelectedIcon(
+            image: 'assets/custom_svg_icons/message.svg',
+            index: 1,
+            label: 'Messages',
+            onTap: onTap,
+          ),
+          _SelectedIcon(
+            image: 'assets/custom_svg_icons/profile.svg',
+            index: 2,
+            label: 'Profile',
+            onTap: onTap,
+          ),
+          _SelectedIcon(
+            image: 'assets/custom_svg_icons/settings.svg',
+            index: 3,
+            label: 'Settings',
+            onTap: onTap,
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _SelectedIcon extends StatelessWidget {
+  const _SelectedIcon({
+    required this.image,
+    required this.index,
+    required this.label,
+    required this.onTap,
+    Key? key,
+  }) : super(key: key);
+
+  final String image;
+  final int index;
+  final String label;
+  final Function onTap;
+
+  @override
+  Widget build(BuildContext context) {
+    const double defaultIconHeight = 20;
+    const double selectedIconHeight = 20;
+
+    return Consumer<TabManager>(
+      builder: (context, tabManager, child) {
+        bool isSelected = tabManager.selectedTab == index;
+
+        return TextButton(
+          style: TextButton.styleFrom(
+            padding: const EdgeInsets.all(0),
+            splashFactory: NoSplash.splashFactory,
+          ),
+          onPressed: () {
+            tabManager.goToTab(index);
+            onTap();
+          },
+          child: Column(
+            mainAxisAlignment:
+                (isSelected) ? MainAxisAlignment.end : MainAxisAlignment.start,
+            children: [
+              const SizedBox(
+                height: 10,
+              ),
+              SvgPicture.asset(
+                image,
+                height: (isSelected) ? selectedIconHeight : defaultIconHeight,
+                color: (isSelected)
+                    ? Theme.of(context).colorScheme.secondary
+                    : Theme.of(context).colorScheme.primary,
+              ),
+              SizedBox(
+                height: (isSelected) ? 17 : 5,
+              ),
+              (isSelected)
+                  ? SvgPicture.asset(
+                      'assets/custom_svg_icons/selection.svg',
+                      color: Theme.of(context).colorScheme.secondary,
+                      fit: BoxFit.cover,
+                    )
+                  : Text(
+                      label,
+                      style: Theme.of(context).textTheme.labelSmall?.copyWith(
```

Comment:
create a theme file something like this
```dart
final themeData = ThemeData(
	fontFamily: 'Poppins',
	accentColor: accentColor,
	primaryColor: primaryColor,
	textTheme: TextTheme(
		headline1: TextStyle(
           color: someColor,
           fontWeight: FontWeight.w700,
           fontSize: 34,
       	),
       	headline2: TextStyle(
         color: accentColor,
         fontWeight: FontWeight.bold,
         fontSize: 16,
       ),
       headline3: TextStyle(
         color: accentColor,
         fontSize: 16,
       ),
	),
);
```
then apply that theme on where ever your MaterialApp is
```dart
MaterialApp(
   title: 'sdfsdf',
   theme: themeData,
   home: SomeWidget(),
)
```

then whenever you're gonna use textTheme make sure you initialize a textTheme inside the build widget
```dart
Widget build(BuildContext context) {
   final textTheme = Theme.of(context).textTheme;

  return Text(
     'some text',
     style: textTheme.headline1, // or whatever style u need to use.
  );
}
```

Reply:
okay sir. ill pass the main text theme to a variable textTheme for every widget. thanks for this